packages/engine-core/src/data-proxy/utils/getClientVersion.ts

```diff
@@ -1,22 +1,34 @@
 import type { EngineConfig } from '../../common/Engine'
+import { NotImplementedYetError } from '../errors/NotImplementedYetError'
 
 /**
  * Determine the client version to be sent to the DataProxy
  * @param config
  * @returns
  */
-export function getClientVersion(config: EngineConfig) {
-  const [version, suffix] = config.clientVersion?.split('-') ?? []
-
-  // we expect the version to match the pattern major.minor.patch
-  if (!suffix && /^[1-9][0-9]*\.[0-9]+\.[0-9]+$/.test(version)) {
-    return version
+export function getClientVersion(config: EngineConfig): string {
+  if (process.env.PRISMA_CLIENT_DATA_PROXY_CLIENT_VERSION) {
+    // should we log with debug here that this env var is used?
+    return process.env.PRISMA_CLIENT_DATA_PROXY_CLIENT_VERSION
+  } else if (config.clientVersion) {
+    const [version, suffix] = config.clientVersion.split('-') ?? []
+    // we expect the version to match the pattern major.minor.patch
+    if (!suffix && /^[1-9][0-9]*\.[0-9]+\.[0-9]+$/.test(version)) {
+      return version
+    } else {
+      throw new NotImplementedYetError('Only `major.minor.patch` versions are supported.', {
+        clientVersion: config.clientVersion,
+      })
+    }
+  } else {
+    throw new NotImplementedYetError(
+      'clientVersion or `PRISMA_CLIENT_DATA_PROXY_CLIENT_VERSION` env var needs to be set with a `major.minor.patch` version.',
+      {
+        clientVersion: '',
+      },
+    )
   }
```

Comment:
Can this ever be triggered or isn't version just the one from pkg.json?

Reply:
In the case of PDP for example, when you call the function directly, clientVersion is optional but is needed for dataproxy, so users should not see this.

Reply:
Ah ok, so if in PDP we _also_ deleted the `clientVersion` param (and not just gave it an actual version and removed `engineVersion` it would be allowed by the code - but then later fail?

Reply:
No they passed a string instead a major.minor.patch but since `clientVersion` is optional I thought having a test for this case could help. The other alternative is to make it required, but would need to check what that impacts.